modules/tinymce/src/core/test/ts/browser/focus/HighlightOnFocusTest.ts

```diff
@@ -54,6 +57,58 @@ describe('browser.tinymce.core.focus.HighlightOnFocus', () => {
       await focusOutsideEditor();
       assertIsNotHighlighted(editor);
     });
+
+    it('TINY-9277: Content area should be highlighted on focus and removed when shifted to menubar', async () => {
+      const editor = hook.editor();
+      assertIsNotHighlighted(editor);
+      editor.focus();
+      assertIsHighlighted(editor);
+      TinyUiActions.clickOnMenu(editor, 'button:contains("Edit")');
+      await TinyUiActions.pWaitForUi(editor, '*[role="menu"]');
+      await Waiter.pWait(0);
```

Comment:
Why the wait here. My flake radar goes off. Should this be a Waiter that waits for the assert to pass instead? Will this fail it it takes say 10ms for the menu to appear. The same goes for all the pWaits in this test can they all be a `await pAssertIsNotHighlighted(editor)` instead?

Reply:
Because the `focusout` has `setTimeout`, but I have tried with `await pAssertIsNotHighlighted(editor)` and that worked. Thank you!